### ISSUE TITLE
Request for assistance: wdv-cli asl contacts-book setup:tests command to set up tests for the contact book project not working

### ISSUE THREAD

**Issue by BradleyMatera** [jworkman/ASL-Knowledge-Base#13]
## **Issue Description**

I am experiencing issues with running the wdv-cli asl contacts-book setup:tests command to set up tests for the contact book project (Advanced Server-side Languages). The course_code is not being recognized, and the script exits with the error course_code: command not found. I have tried multiple solutions including verifying the installation of required packages, manually setting the course code, testing different Node.js versions, and even attempting to run the tests directly using Jest, but the issue persists.

**Key** **Points:**

	•	CLI Issue: The wdv-cli tool appears to be malfunctioning when trying to set the course code for the ASL module.
	•	Missing Tests: The tests are not being downloaded or set up, which prevents running the tests for the contacts book.
	•	Environment Verified: Several attempts were made to run the command with different Node.js versions, verbose output, and a functioning Docker environment.

**Full Error Backtrace**

The initial command that I ran:

wdv-cli asl contacts-book setup:tests

**The output was:**

/Users/bradleymatera/.nvm/versions/node/v16.20.2/bin/wdv-cli: line 134: course_code: command not found
Setting course code to: 
Usage: wdv-cli <course code> <assignment> <action>:<task>

Commands:
  pwa                  Programming for Web Applications (PWA-O | WDV3322)
    car-finder         Car list assignment for week one
      download:data    Downloads all of the JSON lists & data needed for assignment.

  asl                  Advanced Server-side Languages online course (ASL-O | WDV442)
    contacts-book      Space tracker project for weeks three and four of ASL/WDV442
      setup:tests      Runs all of the tests for your contacts book API assignment.

**Next, I tried using the verbose flag for more detailed output:**

wdv-cli asl contacts-book setup:tests --verbose

**The output remained similar:**

/Users/bradleymatera/.nvm/versions/node/v16.20.2/bin/wdv-cli: line 134: course_code: command not found
Setting course code to: 
Usage: wdv-cli <course code> <assignment> <action>:<task>

The error message suggests that somewhere in the script on line 134, the variable course_code is either not declared correctly or not being set. Despite specifying the asl module, it fails to proceed with setting up the tests.

**Additional Information (Things I Tried)**

**1. Docker Setup:**

	•	The Node.js application is running inside Docker containers.
	•	Docker and MongoDB are running correctly as per the following output:

docker ps

**Output:**

CONTAINER ID   IMAGE          COMMAND                  CREATED          STATUS          PORTS                      NAMES
a92562ba20cd   mongo:latest   "docker-entrypoint.s…"   About a minute ago   Up About a minute   0.0.0.0:27017->27017/tcp   restfulroutes_mongodb

**2. Node.js Versions:**

	•	Tried using two different Node.js versions: 16.20.2 and 18.20.4.
	•	NVM was used to switch between these versions using the command:

nvm use 16.20.2

or

nvm use 18.20.4

**3. Installed Jest Manually:**

	•	As the CLI wasn’t functioning, I installed Jest to manually run the tests. The commands **used were:**

npm install jest --save-dev
npm test

**Since the contacts.test.js file was never copied, no tests were found.**

**4. Axios and Other Dependencies:**

	•	Verified that axios and all other required dependencies were installed correctly:

**npm install axios**

	•	Node modules were correctly installed, as verified by the following:

ls -l node_modules

5. Exporting the PATH:

	•	I tried setting the course_code manually using an environment variable:

export course_code="ASL"

**However, this had no effect and the error persisted.**

6. Multiple Attempts to Run the CLI Tool:

	•	Repeated multiple attempts to run the wdv-cli asl contacts-book setup:tests command across both Node.js versions (16.20.2 and 18.20.4), with and without the verbose flag. The issue with course_code persisted.

7. Created a Fresh Directory:

	•	Remade the project directory multiple times using VSCode on Mac, each time ensuring the environment was set up correctly.

Screenshots
<img width="1728" alt="Screenshot 2024-10-08 at 5 22 51 PM" src="https://github.com/user-attachments/assets/4e2ea872-1a1b-49be-84c5-d7de4f215f6f">

<img width="1728" alt="Screenshot 2024-10-08 at 5 36 09 PM" src="https://github.com/user-attachments/assets/2b2a43d3-6ef1-47dc-98e8-c1a0d0cd7d66">

<img width="1728" alt="Screenshot 2024-10-08 at 5 36 44 PM" src="https://github.com/user-attachments/assets/5f7f4d91-ed8c-42c6-b71e-7226aea5d712">

Any Other Notes

Steps to Reproduce:

	1.	Node.js Environment:
	•	Node.js versions used: 16.20.2 and 18.20.4
	•	Installed via NVM and switched between versions using nvm use.
	2.	Installed the WDV CLI Tool Globally:
	•	Command used for global installation of the CLI tool:

npm install -g @jworkman-fs/wdv-cli

	3.	Attempted to Run the Tests Setup:
	•	Attempted to run the test setup command:

wdv-cli asl contacts-book setup:tests

	4.	Error Received:
	•	The error consistently showed that course_code was not being recognized, even with the verbose flag.

Additional Attempts Made:

	•	Manually Installed Jest:
	•	Installed Jest and tried to manually run the tests, but no test files were found due to the setup failure:

npm install jest --save-dev
npm test

	•	Docker Configuration:
	•	The application is running in Docker with MongoDB functioning correctly. However, the test setup cannot proceed due to the CLI tool issue.

Expected Outcome:

	•	Contacts Test File Download: The contacts.test.js file should have been copied into the project folder, allowing the use of Jest with the command npm test to run the provided tests.

**Comment by BradleyMatera** [comment]
TLDR;
	•	Docker setup and verification.
	•	Node.js version switches using NVM.
	•	Installing and manually trying Jest.
	•	Various attempts to run the wdv-cli command with and without the verbose flag.
	•	Exporting the course_code manually.
	•	Creating a fresh directory multiple times.
	•	Installing dependencies like axios.
	•	Repeated attempts across multiple environments

**Comment by jworkman** [comment]
Hi Bradley,

I'm checking on my end now to see if my recent update to the wdv-cli tool broke something. Stand by.

**Comment by jworkman** [comment]
Issue has been resolved and deployed to `v1.4.3` of the `@jworkman-fs/wdv-cli` package. If you have an old version installed please run the following command to update it. 

```npm update -g @jworkman-fs/wdv-cli```

**Comment by jworkman** [comment]
Closing out ticket that has been marked closed/resolved over slack convo.
